ambari-server/src/main/resources/stacks/HDP/2.6/upgrades/config-upgrade.xml

```diff
@@ -53,18 +53,14 @@
       </component>
       <component name="DATANODE">
         <changes>
-          <definition xsi:type="configure" id="hdfs_set_data_transfer_protection"
-                      summary="Enables SASL for authentication of data transfer protocol">
+          <definition xsi:type="configure" id="hdfs_set_data_transfer_protection" summary="Enables SASL for authentication of data transfer protocol">
             <type>hdfs-site</type>
-            <set key="dfs.data.transfer.protection" value="authentication,privacy" if-type="hdfs-site"
-                 if-key="dfs.http.policy" if-value="HTTPS_ONLY"/>
+            <set key="dfs.data.transfer.protection" value="authentication,privacy" if-type="cluster-env" if-key="security_enabled" if-value="true" />
           </definition>
 
-          <definition xsi:type="configure" id="hdfs_set_hadoop_rpc_protection_on_kerberized_cluster"
-                      summary="Encrypting the data transfered between hadoop services and clients">
+          <definition xsi:type="configure" id="hdfs_set_hadoop_rpc_protection_on_kerberized_cluster" summary="Encrypting the data transfered between hadoop services and clients">
             <type>core-site</type>
-            <set key="hadoop.rpc.protection" value="authentication,privacy" if-type="cluster-env"
-                 if-key="security_enabled" if-value="true"/>
+            <set key="hadoop.rpc.protection" value="authentication,privacy" if-type="cluster-env" if-key="security_enabled" if-value="true" />
```

Comment:
I don't think this is correct. Security doesn't guarantee Kerberos. Instead, we have a <condition> element which should be used.

Reply:
You are right; instead I'll check if `hadoop.security.authentication` is set to `kerberos` in `core-site`. Are you OK with this approach?

Reply:
Hm..just reading Nate's comment below and he is right too. We already have the kerberos xsi:security condition in `nonrolling-upgrade-3.0.xml` so that we can simply remove this check here. Any objection?

Reply:
@jonathan-hurley 
I implemented the change I mentioned above; if you had time could you please review? Thanks in advance!